code/modules/mapping/space_management/multiz_helpers.dm

```diff
@@ -34,14 +34,10 @@
 	return get_step_multiz(src, DOWN)
 
 /proc/dir_inverse_multiz(dir)
-	var/holder = dir & (UP|DOWN)
-	if((holder == NONE) || (holder == (UP|DOWN)))
-		return turn(dir, 180)
-	dir &= ~(UP|DOWN)
-	dir = turn(dir, 180)
-	if(holder == UP)
-		holder = DOWN
-	else
-		holder = UP
-	dir |= holder
-	return dir
+	var/vertical = dir & (UP|DOWN)
+	dir &= ~vertical	// remove dir from vertical
+	if(vertical == (UP|DOWN))		// both up/down specified
+		vertical = NONE
+	if(dir)		// byond will be a shit if dir is 0 so let's not have dir be 0.
+		dir = turn(dir, 180)
+	return dir | vertical

```

Comment:
Just use `#define REVERSE_DIR(dir) ( ((dir & 85) << 1) | ((dir & 170) >> 1) )`

Reply:
can you explain how that works i am not that good at binary 😅

Reply:
rohesie spewing out some weird binary magic and prefixes it with "just" kek

Reply:
85 is 0b1010101 or (NORTH|EAST|UP|64) (wait what why is 64 in there?) and 170 is 0b10101010 or (SOUTH|WEST|DOWN|128) (again why is 128 in there thats not a dir) and basically, bitshifting NORTH left gets SOUTH, SOUTH right gets NORTH, same with EAST and WEST, same with UP and DOWN

Reply:
Just an extra digit for safety, you could get away with removing it.
In binary 85 looks like this:
`01010101`
And 170 looks like this:
`10101010`
So first you are doing a bitwise `AND` operation, taking only those relevant bits, and shifting them once. Then you are combining the result.

Example, you have the following direction: `UP | EAST`
In binary that looks like
`00100100`

So first we use the AND operation of this with `85`, so we have this
```
00100100 &
01010101
00000100
```
That result is simply `EAST`, and bitshifting it left we simply get `WEST` (`00001000`)

Now that we did `(dir & 85) << 1)` we do the other part.

Again, `dir & 170)` first
```
00100100 &
10101010
00100000
```
The result is `UP`, and bitshifting it right gets us
`00010000` which is `DOWN`

Now we combine both results through the bitwise OR, and we have `DOWN | WEST`
Since we started with `UP | EAST` we have successfully reverted the direction.

Reply:
Should I just get rid of the proc and switch stuff to use a macro then? If this is the case I don't think we'll need the proc, right?

Reply:
That's the idea!
On TGMC we just run the macro everywhere we want to invert a direction.

Reply:
alright gotcha i'll do this when i'm home, thanks for the lesson on bitflags!